src/program/frame.rs

```diff
@@ -87,6 +87,19 @@ impl FrameSet {
         self.frames.insert(identifier, attributes);
     }
 
+    /// Return a new [FrameSet] which describes only the given [FrameIdentifier]s.
+    pub fn intersection(&self, identifiers: &HashSet<&FrameIdentifier>) -> Self {
+        let mut new_frameset = Self::new();
+
+        for (identifier, definition) in &self.frames {
+            if identifiers.contains(&identifier) {
+                new_frameset.insert(identifier.clone(), definition.clone())
+            }
+        }
+
+        new_frameset
```

Comment:
Perfectly straightforward, but I think it might be more idiomatic as
```rust
FrameSet {
    frames: self
        .frames
        .iter()
        .filter_map(|(id, attr)| {
            identifiers
                .contains(id)
                .then_some((id.clone(), attr.clone()))
        })
        .collect(),
}
```

Reply:
Could arguably do something shorter with `self.frames.retain(…)`, but that might need cloning the entire thing first.

Reply:
To improve on @genos's suggestion, there's also:

```rust
FrameSet {
    frames: self
        .frames
        .iter()
        .filter(|(id, _)| identifiers.contains(id))
        .cloned()
        .collect(),
}
```